### PR TITLE
[Test Modernization]: Add test modernization to  CalloutCard, Caption, CheckableButton, Resizer, VideoThumbnail 

### PR DESCRIPTION
**Pull request by kpinzas-sh**
**WHY are these changes introduced?**

I am updating existing tests for CalloutCard, Caption, CheckableButton, Resizer, VideoThumbnail components to use the new modern framework.

**WHAT is this pull request doing?**

Part of test modernization (https://docs.google.com/spreadsheets/d/1GBuEZbOpVYJLNISK7gL69DU8rCxocn5L5GYaKtPXPbU/edit#gid=1498187033), updating tests using {mountWithAppProvider} from 'test-utilities/legacy' to {mountWithApp} from 'test-utilities'.

### CODE REVIEW

File: src/components/TextField/components/Resizer/tests/Resizer.test.tsx [hunk 1]
```diff
@@ -58,52 +53,47 @@ describe('<Resizer />', () => {
...
   describe('minimumLines', () => {
     it('renders a number of <br> tags equivalent to minimumLines', () => {
       const minimumLines = 3;
-      const resizer = mountWithAppProvider(
+      const resizer = mountWithApp(
         <Resizer {...mockProps} minimumLines={minimumLines} />,
       );
-      const breakingSpaces = findByTestID(resizer, 'MinimumLines');
-      expect(breakingSpaces.html()).toContain('<br><br><br>');
+      expect(resizer.find('div')!.html()).toContain('<br><br><br>');
```
Comment (kpinzas-sh) [inline src/components/TextField/components/Resizer/tests/Resizer.test.tsx:89]: This part is where I have more doubts, not sure if someone can have a suggestion that is better related to this.

Comment (kpinzas-sh) [reply src/components/TextField/components/Resizer/tests/Resizer.test.tsx:89]: I think here would be better to use toContainReactHTML. I will check if that is the correct approach.

**Review by alex-page — APPROVED** [review]

File: src/components/TextField/components/Resizer/tests/Resizer.test.tsx (hunk 1 above)
Comment (rdott) [reply src/components/TextField/components/Resizer/tests/Resizer.test.tsx:89]:
```suggestion
     expect(resizer.find('div'))!. toContainReactHTML('<br><br><br>');
```
Would have been my thoughts on this.

Comment (rdott) [reply src/components/TextField/components/Resizer/tests/Resizer.test.tsx]: My bad, `toContainReactHtml` not `toContainReactHTML`:

```js
expect(resizer.find('div'))!.toContainReactHtml('<br><br><br>');
```

Comment (kpinzas-sh) [reply src/components/TextField/components/Resizer/tests/Resizer.test.tsx]: don't worry! I push it but i need to add it to the other test in the list.

File: src/components/CheckableButton/tests/CheckableButton.test.tsx [hunk 2]
```diff
@@ -18,65 +17,73 @@ describe('<CheckableButton />', () => {
...
 
     it('is called when the CheckableButton pressed with spacebar', () => {
       const spy = jest.fn();
-      const element = mountWithAppProvider(
+      const element = mountWithApp(
         <CheckableButton {...CheckableButtonProps} onToggleAll={spy} />,
       );
-      element.find(Checkbox).first().simulate('click', {
+
+      element.find(Checkbox)!.find('input')!.trigger('onKeyUp', {
```
[the remark this answers is not in the document]
Comment (rdott) [reply src/components/CheckableButton/tests/CheckableButton.test.tsx:83]: My feeling is that there should be a test that checks that the function is passed to the `Checkbox` component, but not testing if it’s called, as this is already tested in` Checkbox.test.tsx` (As discussed on Slack)

**Pull request merged**